update allowReuse nullability

hpoul opened the pull request:
imho allowReuse should either be a non-nullable value or the default value should be used as a fallback when accessed.

otherwise callers can pass in `allowReuse: null` and cause a crash. 😳

isoos commented:
@hpoul: good catch, thank you! I am wondering, and maybe you have an opinion on that: is it good that the two method has a different default value? And if we change them, what would be the good default?

Also: could you please make a note about this in the CHANGELOG and bump the version in pubspec.yaml?

hpoul commented:
rebased, updated `CHANGELOG` and `pubspec.yaml`.

I don't think it's a good idea to have different defaults, especially if it's not documented *why* there is a difference. But it's probably a worse idea to change it 😅️

isoos commented:
Thanks! Should I release this, or do you have something else in the works?

hpoul commented:
@isoos thx, that's it, except you can give me hints on how to fix #11 :) i don't understand why there is a difference between `query` and `execute` in the first place 🤔

isoos commented:
Unfortunately I have no idea on how that code was created and why it differs, I become involved with the package much later.
